engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/AtomConstructor.java

```diff
@@ -317,8 +317,11 @@ public static AtomConstructor accessorFor(Function fn) {
   public static Map<String, RootNode> collectFieldAccessors(EnsoLanguage language, Type type) {
     var constructors = type.getConstructors().values();
     var roots = new TreeMap<String, RootNode>();
-    if (constructors.size() != 1) {
+    if (constructors.size() > 1) {
       var names = new TreeMap<String, List<GetFieldWithMatchNode.GetterPair>>();
+      // We assume that all the constructors have the same definition scope. So we
+      // take just the first one.
```

Comment:
I tried to have an assert for this. But it did not work, because every invocation of `getDefinitionScope()` creates a different object with same contents. Could be verified with a proper `equals` method on `ModuleScope` but that feels like an overkill for this PR. Hence, I have only written this comment here.